core/services/relay/evm/target_strategy.go

```diff
@@ -0,0 +1,226 @@
+		return &writetarget.TransmissionState{
+			Status:      writetarget.TransmissionStateNotAttempted,
+			Transmitter: transmissionInfo.Transmitter.String(),
+			Err:         nil,
+		}, nil
+	case TransmissionStateSucceeded:
+		t.lggr.Infow("returning without a transmission attempt - report already onchain ", "executionID", request.Metadata.WorkflowExecutionID)
+		return &writetarget.TransmissionState{
+			Status:      writetarget.TransmissionStateSucceeded,
+			Transmitter: transmissionInfo.Transmitter.String(),
+			Err:         nil,
+		}, nil
+	case TransmissionStateInvalidReceiver:
+		t.lggr.Infow("returning without a transmission attempt - transmission already attempted, receiver was marked as invalid", "executionID", request.Metadata.WorkflowExecutionID)
+		return &writetarget.TransmissionState{
+			Status:      writetarget.TransmissionStateFatal,
+			Transmitter: transmissionInfo.Transmitter.String(),
+			Err:         ErrTxFailed,
+		}, ErrTxFailed
+	case TransmissionStateFailed:
+		receiverGasMinimum := t.receiverGasMinimum
+		if r.Config.GasLimit != nil {
+			receiverGasMinimum = *r.Config.GasLimit - ForwarderContractLogicGasCost
+		}
+		if transmissionInfo.GasLimit.Uint64() > receiverGasMinimum {
+			t.lggr.Infow("returning without a transmission attempt - transmission already attempted and failed, sufficient gas was provided", "executionID", request.Metadata.WorkflowExecutionID, "receiverGasMinimum", receiverGasMinimum, "transmissionGasLimit", transmissionInfo.GasLimit)
+			return &writetarget.TransmissionState{
+				Status:      writetarget.TransmissionStateFatal,
+				Transmitter: transmissionInfo.Transmitter.String(),
+				Err:         ErrTxFailed,
+			}, ErrTxFailed
+		} else {
+			t.lggr.Infow("non-empty report - transmission should be retried", "request", request, "reportLen", len(r.Inputs.SignedReport.Report), "reportContextLen", len(r.Inputs.SignedReport.Context), "nSignatures", len(r.Inputs.SignedReport.Signatures), "executionID", request.Metadata.WorkflowExecutionID, "receiverGasMinimum", receiverGasMinimum, "transmissionGasLimit", transmissionInfo.GasLimit)
+			return &writetarget.TransmissionState{
+				Status:      writetarget.TransmissionStateFailed,
+				Transmitter: transmissionInfo.Transmitter.String(),
+				Err:         ErrTxFailed,
+			}, ErrTxFailed
+		}
+	}
+
+	return &writetarget.TransmissionState{
+		Status:      writetarget.TransmissionStateFatal,
+		Transmitter: transmissionInfo.Transmitter.String(),
+		Err:         ErrTxFailed,
+	}, fmt.Errorf("unexpected transmission state: %v", transmissionInfo.State)
+}
+
+// TransmitReport constructs the tx to transmit the report, and defines
+// any specific handling for sending the report via ChainWriter.
+func (t *evmTargetStrategy) TransmitReport(ctx context.Context, _ []byte, _ []byte, _ [][]byte, request capabilities.CapabilityRequest) (string, error) {
```

Comment:
What's the point of these unused args in the interface?

Reply:
On Aptos we use them directly, but on EVM we just get these values directly from the decoded report.

Reply:
Ok, so my question is what's the point in doing it in two different ways?

Reply:
On Aptos we have no need to decode the request since we don't need to check a chain specific value like we need to for the gas limit here. For EVM, since we are already decoding the request I decided it was better to get everything from the decoded request since it is more chain specific, but the values should be the same.

Reply:
Well that's not a good way to abstract this logic

Reply:
Agreed, I can add improving this abstraction to the future work